src/main/java/io/supertokens/inmemorydb/queries/PasswordlessQueries.java

```diff
@@ -0,0 +1,612 @@
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, linkCodeHash);
+            ResultSet result = pst.executeQuery();
+            if (result.next()) {
+                return PasswordlessCodeRowMapper.getInstance().mapOrThrow(result);
+            }
+        }
+        return null;
+    }
+
+    public static void deleteCode_Transaction(Start start, Connection con, String codeId) throws SQLException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessCodesTable() + " WHERE code_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, codeId);
+            pst.executeUpdate();
+        }
+    }
+
+    public static void createUser(Start start, UserInfo user)
+            throws StorageTransactionLogicException, StorageQueryException {
+        start.startTransaction(con -> {
+            Connection sqlCon = (Connection) con.getConnection();
+            try {
+                {
+                    String QUERY = "INSERT INTO " + Config.getConfig(start).getUsersTable()
+                            + "(user_id, recipe_id, time_joined)" + " VALUES(?, ?, ?)";
+                    try (PreparedStatement pst = sqlCon.prepareStatement(QUERY)) {
+                        pst.setString(1, user.id);
+                        pst.setString(2, RECIPE_ID.PASSWORDLESS.toString());
+                        pst.setLong(3, user.timeJoined);
+                        pst.executeUpdate();
+                    }
+                }
+
+                {
+                    String QUERY = "INSERT INTO " + Config.getConfig(start).getPasswordlessUsersTable()
+                            + "(user_id, email, phone_number, time_joined)" + " VALUES(?, ?, ?, ?)";
+                    try (PreparedStatement pst = sqlCon.prepareStatement(QUERY)) {
+                        pst.setString(1, user.id);
+                        pst.setString(2, user.email);
+                        pst.setString(3, user.phoneNumber);
+                        pst.setLong(4, user.timeJoined);
+                        pst.executeUpdate();
+                    }
+                }
+                sqlCon.commit();
+            } catch (SQLException throwables) {
+                throw new StorageTransactionLogicException(throwables);
+            }
+            return null;
+        });
+    }
+
+    public static void updateUserEmail_Transaction(Start start, Connection con, String userId, String email)
+            throws SQLException, UnknownUserIdException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getPasswordlessUsersTable()
+                + " SET email = ? WHERE user_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, email);
+            pst.setString(2, userId);
+            if (pst.executeUpdate() != 1) {
+                throw new UnknownUserIdException();
+            }
```

Comment:
See https://github.com/supertokens/supertokens-mysql-plugin/pull/20#discussion_r772914215

Reply:
@porcellus this needs to be updated.